x-pack/test/detection_engine_api_integration/security_and_spaces/tests/alerts/alerts_compatibility.ts

```diff
@@ -161,6 +177,132 @@ export default ({ getService }: FtrProviderContext) => {
         );
         expect(enrichmentProviders).to.eql([expectedProvider, expectedProvider]);
       });
+
+      it('should generate a signal-on-legacy-signal with legacy index pattern', async () => {
+        const rule: ThreatMatchCreateSchema = getThreatMatchRuleForSignalTesting([
+          '.siem-signals-*',
+        ]);
+        const { id } = await createRule(supertest, log, rule);
+        await waitForRuleSuccessOrStatus(supertest, log, id);
+        await waitForSignalsToBePresent(supertest, log, 1, [id]);
+        const signalsOpen = await getSignalsByIds(supertest, log, [id]);
+        expect(signalsOpen.hits.hits.length).greaterThan(0);
```

Comment:
Can we also verify that the alerts are generated in the correct format? I started investigating https://github.com/elastic/kibana/issues/125050 and found that it only happens with alerts-on-legacy alerts, and is due to the alerts being written with the `kibana.*` fields in nested notation copied from the source legacy alert and the `kibana.*` fields are also written in dot notation with the new values for the alert-on-alert doc.

Reply:
Pushed up this fix and additional tests.